packages/appsync/bin/typegen

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env node
+
+const readline = require("readline");
+const events = require("events");
+const fs = require("fs");
+
+const apiPath = process.argv[2];
+
+if (!apiPath) {
+  throw new Error("Pass path to api.ts");
+}
+
+const path = require("path");
+
+const FILE_PATH = path.join(process.cwd(), apiPath);
+
+const QUERY_REGEX = /export type (((List|Get|Sync).*)Query) =/;
```

Comment:
This would not work for custom queries, mutations and subscriptions. Perhaps we should extract the names from the graphql schema and then extract them accordingly from the codegenerated types files?

Reply:
maybe we should do this as a separate task tho

Reply:
yes this won't work if the custom queries don't follow the same naming format. we can do improvements on it on a separate task.